### DISCUSSION TITLE
Double Y axis homing problem

### DISCUSSION THREAD

**Discussion by MarCiu11** [fra589/grbl-Mega-5X#273] [General]
I have a problem with the Y axis motors when homing **$22=1**.
- one motor is connected as AX 2, the other as AX 5.
When homing is off **$22=0**. Everything is fine, both motors move smoothly and the gate moves as expected.
But when I turn on homing **$22=1**, one motor spins forward (endstop), and the other in the opposite direction and the gate jams - it's diagonally.

My settings in the file "config.h":
...
_#define N_AXIS 5 // Number of axes (3 to 6)_
...
_#elif N_AXIS == 5 // 5 axis: homing
   #define HOMING_CYCLE_0 (1 << AXIS_3)
   #define HOMING_CYCLE_1 (1 << AXIS_1)
   #define HOMING_CYCLE_2 ((1 << AXIS_2) | (1 << AXIS_5))_
...
First Z axis, then X axis, then both Y axes.

Both endstops are on the Y + axis (at the end of the axis).
I have the $23 parameter set to 0 (**$23=0** So all switches are set to **+**)
Unfortunately, one engine at HOME goes to the front, the other to the back.

Anything else needs to be set? What am I doing wrong?
Please help :)

**Comment by fra589** [comment]
Hello @MarCiu11,  
  
Please, also post the full $$ and $I result.  
  
@++;  
Gauthier.

**Comment by MarCiu11** [comment]
Thank you for your willingness to help.
Full details below:

$I
[VER:1.2e.20220303:]
[AXS:5:XYZAY]
[OPT:VNMGH,35,255,0]
ok

$$
$0=10
$1=254
$2=0
$3=0
$4=0
$5=0
$6=0
$10=1
$11=0.020
$12=0.002
$13=0
$20=1
$21=0
$22=1
$23=0
$24=100.000
$25=500.000
$26=250
$27=2.000
$30=12000
$31=550
$32=0
$100=200.000
$101=200.000
$102=400.000
$103=200.000
$104=200.000
$110=500.000
$111=800.000
$112=500.000
$113=500.000
$114=800.000
$120=50.000
$121=50.000
$122=50.000
$123=50.000
$124=50.000
$130=240.000
$131=310.000
$132=92.000
$133=360.000
$134=180.000
ok

**Comment by fra589** [comment]
When using cloned axes, they must have exactly the same configuration.
I haven't checked in the code if this can be the cause, but in your configuration $131 is not equal to $134...

Next, and more importantly, the linear axes must appear first in the numbering before the rotary axes. Your Y clone being linear and the A rotating, you should invert the A axis with the Y clone.
Which gives in config.h:
```
#define N_AXIS 5 // Number of axes (3 to 6)
#define N_AXIS_LINEAR 4 // Number of linear axes, must be <= N_AXIS
// Axis indexing and names
#define AXIS_1 0 // Axis indexing value. Must start with 0 and be continuous.
#define AXIS_1_NAME 'X' // Axis names must be in X, Y, Z, A, B, C, U, V, W, D, E & H.
#define AXIS_2 1
#define AXIS_2_NAME 'Y'
#define AXIS_3 2
#define AXIS_3_NAME 'Z'
#if N_AXIS <3
  #error "N_AXIS must be >= 3. N_AXIS < 3 is not implemented."
#endif
#if N_AXIS > 3
  #define AXIS_4 3
  #define AXIS_4_NAME 'Y' // Letter of axis number 4
#endif
#if N_AXIS > 4
  #define AXIS_5 4
  #define AXIS_5_NAME 'A' // Letter of axis number 5
#endif
```

@++;
Gauthier.

**Comment by MarCiu11** [comment]
Unfortunately, the proposed changes did not help.
Home is ok, but when it gives you manual control (arrows), the Y2 motor is stationary - it does not move at all (with Home it moves in the Y + direction).

I corrected the command $ 131 and $ 134 - now they are identical.

My current configuration:

//----------------------------------------------------------------------
#define N_AXIS 5        // Number of axes (3 to 6)
#define N_AXIS_LINEAR 4 // Number of linears axis, must be <= N_AXIS
// Axis indexing and names
#define AXIS_1 0        // Axis indexing value. Must start with 0 and be continuous.
#define AXIS_1_NAME 'X' // Axis names must be in X, Y, Z, A, B, C, U, V, W, D, E & H.
#define AXIS_2 1
#define AXIS_2_NAME 'Y'
#define AXIS_3 2
#define AXIS_3_NAME 'Z'
#if N_AXIS <3
  #error "N_AXIS must be >= 3. N_AXIS < 3 is not implemented."
#endif
#if N_AXIS > 3
  #define AXIS_4 3
  #define AXIS_4_NAME 'Y' // Letter of axis number 4
#endif
#if N_AXIS > 4
  #define AXIS_5 4
  #define AXIS_5_NAME 'A' // Letter of axis number 5
#endif
#if N_AXIS > 5
  #define AXIS_6 5
  #define AXIS_6_NAME 'C' // Letter of axis number 6
#endif
#if N_AXIS > 6
  #error "N_AXIS must be <= 6. N_AXIS > 6 is not implemented."
#endif
//----------------------------------------------------------------------

**Comment by fra589** [comment]
Hello @MarCiu11,  
  
If you have made changes in the axis configuration, dit you issue the "magic" Grbl command: `$RST=*` to reset the flash configuration ?  
If no, Grbl generally have unpredictable result...

@++;  
Gauthier.

**Comment by fra589** [comment]
Hello @MarCiu11,  
  
Have you made any change in cpu_map.h ?  
if no, may be you have an hardware problem...
  
@++;  
Gauthier.
